File: javascript/matching-brackets/matching-brackets.js
const PAIRS = {"(": ")", "[": "]", "{": "}"}

export const isPaired = (input) => {
    let stack = [];

    for (const char of input)
        if ("([{".includes(char))
            stack.push(char);
        else if (")]}".includes(char))
            if (char != PAIRS[stack.pop()])
                return false;

    return stack.length == 0;
};

File: csharp/matching-brackets/MatchingBrackets.cs
using System;
using System.Collections.Generic;

public static class MatchingBrackets
{
    private static Dictionary<char, char> pairs = new Dictionary<char, char>() {
        {'(', ')'},
        {'[', ']'},
        {'{', '}'}
    };

    public static bool IsPaired(string input)
    {
        var s = new Stack<char>();

        foreach (var c in input) {
            if (c == '(' || c == '[' || c == '{') {
                s.Push(c);
            } else if (c == ')' || c == ']' || c == '}') {
                if (s.Count == 0)
                    return false;

                if (pairs[s.Pop()] != c)
                    return false;
            }
        }

        return s.Count == 0;
    }
}
